flink-table-store-core/src/main/java/org/apache/flink/table/store/file/writer/AppendOnlyWriter.java

```diff
@@ -116,9 +117,15 @@ public List<DataFileMeta> close() throws Exception {
         return result;
     }
 
+    private FileWriter.Factory<RowData, Metric> createFileWriterFactory() {
```

Comment:
It is already a factory, we can just init the factory in the constructor of this class.